vaadin-list-mixin.html

```diff
@@ -208,15 +208,22 @@
       return -1;
     }
 
+    _isItemHidden (item) {
+      if (getComputedStyle(item).getPropertyValue('display') === 'none') {
+        return true;
+      }
+      return false;
+    }
+
     _setFocusable(idx) {
-      idx = this._getAvailableIndex(idx, 1, item => !item.disabled);
+      idx = this._getAvailableIndex(idx, 1, item => !(item.disabled || this._isItemHidden(item)));
       const item = this.items[idx] || this.items[0];
       this.items.forEach(e => e.tabIndex = e === item ? 0 : -1);
     }
 
     _focus(idx) {
       const item = this.items[idx];
-      this.items.forEach(e => e.focused = e === item);
+      this.items.forEach(e => e.focused = (e === item && !(this._isItemHidden(e) || e.disabled)));
```

Comment:
This change might not be needed (if the condition is already checked elsewhere so you can never reach this block with a hidden item index). If there are no other side effects, it might actually make more sense to have all the validity checks here instead and remove the other checks.

Reply:
done. I removed the checking condition on _focus(idx).